lib/number_format.js

```diff
@@ -949,12 +949,13 @@ var NumberFormat = function (_React$Component) {
       setTimeout(function () {
         var el = e.target;
         var selectionStart = el.selectionStart,
+            selectionEnd = el.selectionEnd,
             _el$value3 = el.value,
             value = _el$value3 === undefined ? '' : _el$value3;
 
 
         var caretPosition = _this4.correctCaretPosition(value, selectionStart);
-        if (caretPosition !== selectionStart) {
+        if (caretPosition !== selectionStart && !(selectionStart === 0 && selectionEnd === value.length)) {
```

Comment:
Can you add a comment on code for `selectionStart === 0 && selectionEnd === value.length` condition.